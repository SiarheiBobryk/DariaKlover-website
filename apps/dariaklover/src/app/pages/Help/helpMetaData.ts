import { PageMetaData } from '@dariaklover/types';

const helpMetaData: PageMetaData = {
  title: 'Помощь',
  heading: 'Помощь',
  path: '/help',
};

export default helpMetaData;
