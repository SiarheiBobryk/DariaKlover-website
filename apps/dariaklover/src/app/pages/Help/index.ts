export { default } from './Help';

export { default as helpMetaData } from './helpMetaData';
