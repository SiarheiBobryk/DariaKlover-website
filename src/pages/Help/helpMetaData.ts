export interface HelpMetaData {
  title: string;
  heading: string;
  path: string;
}

const howToBookMetaData: HelpMetaData = {
  title: 'Помощь',
  heading: 'Помощь',
  path: '/help',
};

export default howToBookMetaData;
