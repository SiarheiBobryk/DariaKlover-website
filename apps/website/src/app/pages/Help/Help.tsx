import * as React from 'react';
import { Helmet } from 'react-helmet';
import { NavLink as RouterNavLink } from 'react-router-dom';

import Link from '@mui/material/Link';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';

import AlertHelp from '../../components/AlertHelp';
import Heading from '../../components/Heading';
import faqMetaData from '../Faq/faqMetaData';
import howToBookMetaData from '../HowToBook/howToBookMetaData';

import helpMetaData from './helpMetaData';

function Help() {
  return (
    <>
      <Helmet>
        <title>{helpMetaData.title}</title>
      </Helmet>
      <Heading>{helpMetaData.heading}</Heading>
      <List>
        <ListItem>
          <Link component={RouterNavLink} to={faqMetaData.path}>
            {faqMetaData.heading}
          </Link>
        </ListItem>
        <ListItem>
          <Link component={RouterNavLink} to={howToBookMetaData.path}>
            {howToBookMetaData.heading}
          </Link>
        </ListItem>
      </List>
      <AlertHelp />
    </>
  );
}

export default Help;
