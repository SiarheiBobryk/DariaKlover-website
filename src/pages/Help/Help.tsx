import * as React from 'react';
import { Helmet } from 'react-helmet';
import { NavLink as RouterNavLink } from 'react-router-dom';

import { styled } from '@mui/material/styles';
import Alert from '@mui/material/Alert';
import AlertTitle from '@mui/material/AlertTitle';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemText from '@mui/material/ListItemText';

import Link from '@mui/material/Link';
import { AppConfigContext, Config } from '../../providers/AppConfigProvider';
import Heading from '../../components/Heading';
import helpMetaData from './helpMetaData';
import howToBookMetaData from '../HowToBook/howToBookMetaData';
import LinkBlank from '../../components/LinkBlank';

export const PUBLIC_PATH = 'responsive_images';

export const Img = styled('img')(function styleImg({ theme }) {
  return {
    maxWidth: '100%',
    borderRadius: theme.typography.pxToRem(8),
  };
});

export const Figure = styled('figure')(function styleFigure({ theme }) {
  return {
    marginTop: 0,
    marginRight: 0,
    marginBottom: theme.typography.pxToRem(8),
    marginLeft: 0,
  };
});

function Help() {
  const {
    instagram: { href: instagramHref },
    telegram: { href: telegramHref },
    whatsapp: { href: whatsappHref },
    email,
  }: Config = React.useContext(AppConfigContext);

  return (
    <>
      <Helmet>
        <title>{helpMetaData.title}</title>
      </Helmet>
      <Heading>{helpMetaData.heading}</Heading>

      <List>
        <ListItem disableGutters>
          <ListItemText
            primary={
              <Link component={RouterNavLink} to={howToBookMetaData.path}>
                {howToBookMetaData.heading}
              </Link>
            }
          />
        </ListItem>
      </List>
      {/* TODO: Move that to the separate component */}
      <Alert severity="info">
        <AlertTitle>Связаться с нами</AlertTitle>
        Если у вас остались вопросы или вы продолжаете испытывать сложности в работе с сайтом, то не стесняйтесь и
        свяжитесь с нами через{' '}
        <LinkBlank href={instagramHref} rel="noopener">
          Instagram
        </LinkBlank>
        ,{' '}
        <LinkBlank href={telegramHref} rel="noopener">
          Telegram
        </LinkBlank>
        ,{' '}
        <LinkBlank href={whatsappHref} rel="noopener">
          WhatsApp
        </LinkBlank>{' '}
        или по{' '}
        <LinkBlank href={`mailto:  ${email}`} rel="noopener">
          электронной почте
        </LinkBlank>
        . Мы обязательно поможем.
      </Alert>
    </>
  );
}

export default Help;
