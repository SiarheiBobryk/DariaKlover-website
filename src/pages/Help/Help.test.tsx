import * as React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { axe, toHaveNoViolations } from 'jest-axe';
import { render, RenderResult } from '@testing-library/react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import Help from './Help';

expect.extend(toHaveNoViolations);

describe('<Help />', () => {
  describe('a11y', () => {
    it('should match a11y standards', async () => {
      const { container }: RenderResult = render(
        <BrowserRouter>
          <Help />
        </BrowserRouter>,
      );

      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });

  describe('snapshotting', () => {
    it('should render correctly', () => {
      const howToBook: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer
        .create(
          <BrowserRouter>
            <Help />
          </BrowserRouter>,
        )
        .toJSON();
      expect(howToBook).toMatchSnapshot();
    });
  });
});
