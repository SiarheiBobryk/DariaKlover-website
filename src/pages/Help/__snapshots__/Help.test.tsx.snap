// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Help /> snapshotting should render correctly 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
  >
    Помощь
  </h2>,
  <ul
    className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      className="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
      disabled={false}
    >
      <div
        className="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
            href="/help/how-to-book"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            Как записаться на консультацию
          </a>
        </span>
      </div>
    </li>
  </ul>,
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-12mvehg-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="InfoOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      <div
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
      >
        Связаться с нами
      </div>
      Если у вас остались вопросы или вы продолжаете испытывать сложности в работе с сайтом, то не стесняйтесь и свяжитесь с нами через 
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="https://t.me/dariaklover"
        onBlur={[Function]}
        onFocus={[Function]}
        rel="noopener"
        target="_blank"
      >
        Telegram
         
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1xzgpfl-MuiSvgIcon-root"
          data-testid="OpenInNewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
          />
        </svg>
      </a>
      , 
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="https://wa.me/+375447841051"
        onBlur={[Function]}
        onFocus={[Function]}
        rel="noopener"
        target="_blank"
      >
        WhatsApp
         
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1xzgpfl-MuiSvgIcon-root"
          data-testid="OpenInNewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
          />
        </svg>
      </a>
       или по
       
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="mailto: dariaklover@gmail.com"
        onBlur={[Function]}
        onFocus={[Function]}
        rel="noopener"
        target="_blank"
      >
        электронной почте (
        <i>
          Email
        </i>
        )
         
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1xzgpfl-MuiSvgIcon-root"
          data-testid="OpenInNewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
          />
        </svg>
      </a>
      . Мы обязательно поможем.
    </div>
  </div>,
]
`;
